### PR TITLE
Add documentation for `ClimateMachine` settings

### LINKED ISSUE #1300: --vtk and output/ creation
Adding a quick note in case someone else runs into this....

The trigger for creating "output/" is having --vtk or --diagnostics set to something other than "never" on CLI (or in ENV). This can be a little confusing because you can request vtk to try and write from function calls in code e.g.

```
vtk_interval = ceil(Int64, timeout / solver_config.dt)
ClimateMachine.Settings.vtk = "$(vtk_interval)steps"
```

from ( https://github.com/CliMA/ClimateMachine.jl/blob/978bc60a5c9d03a240967e7e04733038010bec50/experiments/OceanBoxGCM/ocean_gyre.jl#L91 )

These lines in code no longer trigger the creation of output/, so unless a --vtk SOMETHING CLI option (or ENV) is given then an error occurs e.g.

```
  SystemError: opening file "output/ocean_gyre_mpirank0000_num0001.vtu": No such file or directory
```

(unless the directory output/ already exists).

### PR DESCRIPTION
**Pull request by kpamnany**
# Description

And how they can be controlled.

Closes #1300.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [X] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [X] There are no open pull requests for this already
- [X] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: docs/src/GettingStarted/RunningClimateMachine.md [hunk 1]
```diff
@@ -28,27 +28,42 @@ $ julia --project experiments/AtmosGCM/heldsuarez.jl
...
-(e.g. `CLIMATEMACHINE_SETTINGS_DIAGNOSTICS=default`).
-
-## [Command line arguments](@id ClimateMachine-args)
-
-The `ClimateMachine` allows some aspects of its behavior to be controlled
-by command line arguments. Many drivers make use of this facility
-(but some do not). The command line options recognized can be examined
-interactively, for example:
+Diagnostics-vars) that are written to NetCDF files at specified intervals by
+the `ClimateMachine` when configured to do so by a driver.
```
Comment (charleskawczynski) [inline docs/src/GettingStarted/RunningClimateMachine.md:36]:
```suggestion
Diagnostics-vars) that are written to NetCDF files at user-specified intervals
in the `ClimateMachine` driver `Callbacks`.
```
Maybe include a ref to the callbacks / driver?

Comment (charleskawczynski) [reply docs/src/GettingStarted/RunningClimateMachine.md:36]: The suggestion is just a suggestion. I'm not sure if what I wrote is more transparent, but I think something along those lines would be clear/helpful

File: docs/src/GettingStarted/RunningClimateMachine.md [hunk 2]
```diff
@@ -28,27 +28,42 @@ $ julia --project experiments/AtmosGCM/heldsuarez.jl
...
+1. [Command line arguments](@ref ClimateMachine-clargs) have the highest
+   precedence, but it is possible for a driver to disable parsing of command
+   line arguments. In such a case, only the next two ways can be used to change
+   settings.
+
+2. [Environment variables](@ref ClimateMachine-envvars) are considered after
+   command line arguments.
+
+3. [Programmatic settings](@ref ClimateMachine-kwargs) have the lowest
+   precedence.
```
Comment (charleskawczynski) [inline docs/src/GettingStarted/RunningClimateMachine.md:60]: ❤️ 
I can see using this as a common reference to new users. Looks great.

File: docs/src/GettingStarted/RunningClimateMachine.md [hunk 3]
```diff
@@ -28,27 +28,42 @@ $ julia --project experiments/AtmosGCM/heldsuarez.jl
...
+checkpointing/restarting, etc. There are 3 ways in which these settings can be
+changed:
+
+1. [Command line arguments](@ref ClimateMachine-clargs) have the highest
+   precedence, but it is possible for a driver to disable parsing of command
+   line arguments. In such a case, only the next two ways can be used to change
+   settings.
+
+2. [Environment variables](@ref ClimateMachine-envvars) are considered after
+   command line arguments.
```
Comment (charleskawczynski) [inline docs/src/GettingStarted/RunningClimateMachine.md:57]:
```suggestion
2. [Environment variables](@ref ClimateMachine-envvars) take precedence after
   command line arguments.
```

File: docs/src/GettingStarted/RunningClimateMachine.md [hunk 4]
```diff
@@ -28,27 +28,42 @@ $ julia --project experiments/AtmosGCM/heldsuarez.jl
...
+
+3. [Programmatic settings](@ref ClimateMachine-kwargs) have the lowest
+   precedence.
+
+### [Command line arguments](@id ClimateMachine-clargs)
+
+If a driver configures the `ClimateMachine` to parse command line
+arguments (most do), you can query the list of arguments understood,
```
Comment (leios) [inline docs/src/GettingStarted/RunningClimateMachine.md:65]:
```suggestion
arguments by setting `parse_clargs = true` in `ClimateMachine.init()`, you can query the list of arguments understood,
```
I think there should be some indication for developers on how to turn this flag on.

File: docs/src/GettingStarted/RunningClimateMachine.md (hunk 2 above)
Comment (jakebolewski) [reply docs/src/GettingStarted/RunningClimateMachine.md:60]: @kpamnany ENV variables have the lowest precedence

Comment (kpamnany) [reply docs/src/GettingStarted/RunningClimateMachine.md:60]: Doh! You're right...

**Pull request merged**